src/aarch64/ffi.c

```diff
@@ -731,9 +855,19 @@ ffi_call_int (ffi_cif *cif, void (*fn)(void), void *orig_rvalue,
   else if (flags & AARCH64_RET_NEED_COPY)
     rsize = 16;
 
+#ifdef __CHERI_PURE_CAPABILITY__
+  stack_argsize = arg_size(cif, stack_bytes, isvariadic);
+  stack_padding = __builtin_cheri_round_representable_length(stack_argsize) -
+    stack_argsize;
```

Comment:
Defer the computation of `stack_argsize` and just do the following
```
stack_padding = __builtin_cheri_round_representable_length(stack_bytes) - stack_argsize;
```

Reply:
The current approach does this.